src/jit/importer.cpp

```diff
@@ -1918,39 +1918,67 @@ GenTreePtr Compiler::getRuntimeContextTree(CORINFO_RUNTIME_LOOKUP_KIND kind)
     return ctxTree;
 }
 
-/*****************************************************************************/
-/* Import a dictionary lookup to access a handle in code shared between
-   generic instantiations.
-   The lookup depends on the typeContext which is only available at
-   runtime, and not at compile-time.
-   pLookup->token1 and pLookup->token2 specify the handle that is needed.
-   The cases are:
-
-   1. pLookup->indirections == CORINFO_USEHELPER : Call a helper passing it the
-      instantiation-specific handle, and the tokens to lookup the handle.
-   2. pLookup->indirections != CORINFO_USEHELPER :
-      2a. pLookup->testForNull == false : Dereference the instantiation-specific handle
-          to get the handle.
-      2b. pLookup->testForNull == true : Dereference the instantiation-specific handle.
-          If it is non-NULL, it is the handle required. Else, call a helper
-          to lookup the handle.
- */
+//------------------------------------------------------------------------
+// impRuntimeLookupToTree: return a tree node for a runtime lookup
+//
+// Arguments:
+//   pResolvedToken - resolved token to look up
+//   pLookup - lookup information from runtime
+//   compileTimeHandle - compile time handle for the lookup
+//
+// Returns:
+//   Tree for the lookup. Structure of the tree can vary depending on
+//   the details of the lookup and the jit's code generation mode.
+//
+// Notes:
+//   Imports a dictionary lookup to access a handle in code shared
+//   between generic instantiations. The lookup depends on the
+//   typeContext which is only available at runtime, and not at
+//   compile-time.  pLookup->token1 and pLookup->token2 specify the
+//   handle that is needed.
+//
+//   The cases are:
+//
+//   1. pLookup->indirections == CORINFO_USEHELPER : Call a helper passing it the
+//      instantiation-specific handle, and the tokens to lookup the handle.
+//      Also used if possible when optimizing jit code size or jit time, and
+//      the inline sequences below will result in larger code or slower jitting.
+//
+//   2. pLookup->indirections != CORINFO_USEHELPER :
```

Comment:
I did add something to the "1" entry that was supposed to cover the subsequent cases. Let me call it out better.